### PR TITLE
[FEATURE][symbology] Add a font style setting for font markers

### PR DESCRIPTION
**Pull request by nirvn**
## Description

It's about time our font marker gets a font style name setting:
Screenshot from 2020-03-19 18-38-34

This unlocks rendering of non-default style for a bunch of font families out there. There's not much to say beyond the above-attached screenshot.

@nyalldawson , review appreciated.

### CODE REVIEW

File: src/core/symbology/qgsmarkersymbollayer.cpp [hunk 1]
```diff
@@ -2984,6 +2985,8 @@ QgsSymbolLayer *QgsFontMarkerSymbolLayer::create( const QgsStringMap &props )
 
   QgsFontMarkerSymbolLayer *m = new QgsFontMarkerSymbolLayer( fontFamily, string, pointSize, color, angle );
 
+  if ( props.contains( QStringLiteral( "font_style" ) ) )
+    m->setFontStyle( props[QStringLiteral( "font_style" )] );
```
Comment (nyalldawson) [inline src/core/symbology/qgsmarkersymbollayer.cpp]: Can you checkout how the text renderer stores this? From memory there's some trickiness involving translation.

File: src/gui/symbology/qgssymbollayerwidget.cpp [hunk 2]
```diff
@@ -3525,6 +3550,39 @@ void QgsFontMarkerSymbolLayerWidget::mStrokeWidthUnitWidget_changed()
   }
 }
 
+void QgsFontMarkerSymbolLayerWidget::populateFontStyleComboBox()
+{
+  mFontStyleComboBox->clear();
```
Comment (nyalldawson) [inline src/gui/symbology/qgssymbollayerwidget.cpp]: Eh, surprised we don't have a reusable widget for this..!

File: src/gui/symbology/qgssymbollayerwidget.h [hunk 3]
```diff
@@ -976,6 +976,12 @@ class GUI_EXPORT QgsFontMarkerSymbolLayerWidget : public QgsSymbolLayerWidget, p
 
   public slots:
     void setFontFamily( const QFont &font );
+
+    /**
+     * Sets the font \a style.
+     * \since QGIS 3.14
+     */
+    void setFontStyle( const QString &style );
```
Comment (nyalldawson) [inline src/gui/symbology/qgssymbollayerwidget.h]: Make private

File: tests/src/core/testqgsfontmarker.cpp [hunk 4]
```diff
@@ -134,6 +135,20 @@ void TestQgsFontMarkerSymbol::fontMarkerSymbol()
...
 
+void TestQgsFontMarkerSymbol::fontMarkerSymbolStyle()
+{
+  mReport += QLatin1String( "<h2>Font marker symbol style layer test</h2>\n" );
+
+  mFontMarkerLayer->setColor( Qt::blue );
+  QFont font = QgsFontUtils::getStandardTestFont( QStringLiteral( "Bold" ) );
+  mFontMarkerLayer->setFontFamily( font.family() );
+  // It actually doesn't change the rendering as the QGIS test fonts don't have styles, but test still insures rendering occurs without a problem
+  mFontMarkerLayer->setFontStyle( QStringLiteral( "Italic" ) );
```
Comment (nyalldawson) [inline tests/src/core/testqgsfontmarker.cpp]: It does, it's called "oblique"

**Comment by nyalldawson** [comment]
Nice feature 😁

**Comment by nirvn** [comment]
@nyalldawson , comments addressed.

**Review by nyalldawson — APPROVED** [review]
Looks great!

**Comment by nirvn** [comment]
@nyalldawson , do you know why windows azure fails here? Makes no sense to me.

**Comment by roya0045** [comment]
> @nyalldawson , do you know why windows azure fails here? Makes no sense to me.

Could you add a " at the end of the 4th line of the pipeline yml file just to see if this solves this? I'm not sure where the escape character could be.

**Comment by suricactus** [comment]
Amazing! I was planing to do so these days :) 
Is it working with save/load/copy styles?

**Comment by nirvn** [comment]
Unrelated travis failure (which had been green prior to it going busted), merging.

**Pull request merged**